### ISSUE TITLE
Changing CassandraCluster persistence settings should not be allowed

### ISSUE THREAD

**Issue by wallrj** [jetstack/navigator#233]
~~Right now, if you change the persistence settings for a running cluster, the navigator controller will attempt to update the statefulset with the new volumeclaimtemplate settings.~~

~~I'm not sure whether this is supported and even if so, Navigator doesn't yet have a way to control the rollout of the updated statefulset.~~

We should prevent the user from changing the persistence setting for now.

See:
 * https://github.com/kubernetes/kubernetes/issues/41015
 * https://github.com/kubernetes/kubernetes/pull/46669
 * But https://github.com/kubernetes/community/pull/503 suggests that updating storage settings was out of scope.
 * https://github.com/kubernetes/kubernetes/issues/35941

/kind bug

**Comment by wallrj** [comment]
This shouldn't happen after #256 but we can add validation to make the persistence setting immutable.
Similar to what @kragniz did for Rack and DC
